specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-09-02-preview/fleets.json

```diff
@@ -0,0 +1,854 @@
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/AzureEntityResource"
+        }
+      ]
+    },
+    "FleetMemberProperties": {
+      "description": "Properties of a Fleet member.",
+      "type": "object",
+      "properties": {
+        "clusterResourceId": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "The ARM resource id of the cluster that joins the Fleet. Must be a valid Azure resource id. e.g.: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{clusterName}'."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FleetMemberProvisioningState",
+            "modelAsString": true
+          },
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Joining",
+            "Leaving",
+            "Updating"
+          ],
+          "description": "The provisioning state of the last accepted operation."
+        }
+      }
+    },
+    "FleetCredentialResults": {
+      "type": "object",
+      "properties": {
+        "kubeconfigs": {
+          "type": "array",
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/FleetCredentialResult"
+          },
+          "description": "Base64-encoded Kubernetes configuration file."
+        }
+      },
+      "description": "The list credential result response."
+    },
+    "FleetCredentialResult": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the credential."
+        },
+        "value": {
+          "type": "string",
+          "format": "byte",
+          "readOnly": true,
+          "description": "Base64-encoded Kubernetes configuration file."
+        }
+      },
+      "description": "The credential result response."
+    }
+  },
+  "parameters": {
+    "FleetNameParameter": {
+      "name": "fleetName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 63,
+      "pattern": "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$",
+      "description": "The name of the Fleet resource.",
+      "x-ms-parameter-location": "method"
+    },
+    "FleetMemberNameParameter": {
+      "name": "fleetMemberName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 50,
+      "pattern": "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$",
+      "description": "The name of the Fleet member resource.",
+      "x-ms-parameter-location": "method"
+    },
+    "ClusterNameParameter": {
```

Comment:
>ClusterNameParameter [](http://example.com/codeflow?start=5&length=20)

Is this defined already in

  - Microsoft.ContainerService/preview/2022-09-02-preview/managedClusters.json

?
can you reference that existing definition instead of duplicating?

Reply:
Hi Tim, thanks for the pro-active review.
The only change in this entire PR is a regex validation.
The rest is unchanged from 07-02-preview for fleet resources.